homeassistant/components/aurora/__init__.py

```diff
@@ -1 +1,142 @@
 """The aurora component."""
+
+import asyncio
+from datetime import timedelta
+import logging
+
+from auroranoaa import AuroraForecast
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    AURORA_API,
+    CONF_THRESHOLD,
+    COORDINATOR,
+    DEFAULT_POLLING_INTERVAL,
+    DEFAULT_THRESHOLD,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["binary_sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Aurora component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Aurora from a config entry."""
+
+    conf = entry.data
+    options = entry.options
+
+    session = aiohttp_client.async_get_clientsession(hass)
+    api = AuroraForecast(session)
+
+    longitude = conf[CONF_LONGITUDE]
+    latitude = conf[CONF_LATITUDE]
+    polling_interval = options.get(CONF_SCAN_INTERVAL, DEFAULT_POLLING_INTERVAL)
+    threshold = options.get(CONF_THRESHOLD, DEFAULT_THRESHOLD)
+    name = conf[CONF_NAME]
+
+    try:
+        await api.get_forecast_data(longitude, latitude)
+    except ConnectionError:
+        return False
```

Comment:
A connection error can happen when the internet is not connected. Seems like an excellent use case to be retried (and so raise ConfigEntryNotReady). Why filter it out ?

Reply:
Good call - addressed.